ampd/src/metrics.rs

```diff
@@ -0,0 +1,108 @@
+use std::net::SocketAddr;
+use std::time::{Duration, Instant};
+
+use lazy_static::lazy_static;
+use prometheus::{
+    self, register_histogram, register_int_counter, register_int_gauge, Encoder, Histogram,
+    HistogramOpts, IntCounter, IntGauge, TextEncoder,
+};
+use tokio::task::JoinHandle;
+use warp::Filter;
+
+lazy_static! {
+    // times it calls bond_verifiers
+    pub static ref VERIFIERS_BONDED: IntCounter = register_int_counter!(
+        "verifiers_bonded_calls_total",
+        "Total number times calling bond_verifiers command"
+    ).unwrap();
+
+     // times it calls verifier_address
+    pub static ref VERIFIER_ADDRESS_CALLS: IntCounter = register_int_counter!(
+        "verifier_address_calls_total",
+        "Total number of times calling the verifier_address command"
+    ).unwrap();
+
+    // time it takes to load configuartion file
+    pub static ref CONFIG_LOAD_TIME: Histogram = register_histogram!(
+        HistogramOpts::new(
+            "config_load_seconds",
+            "Time taken to load and parse configuration files"
+        )
+        .buckets(vec![0.005, 0.01, 0.05, 0.1, 0.5, 1.0])
+    ).unwrap();
+    // 2-secnd updates
+    pub static ref UPTIME_SECONDS: IntGauge = register_int_gauge!(
+        "ampd_uptime_seconds",
+        "Number of seconds the application has been running"
+    ).unwrap();
+}
+
+lazy_static! {
+    static ref START_TIME: Instant = Instant::now();
+}
```

Comment:
None of these should be done with `lazy_static`, but as part of the server initialization. The server should wrap the metrics, and then other code update the metrics via a `MetricsClient`, which should be clonable. `MetricsClient` update the `Metrics` via a `mpsc::channel`.